orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -167,25 +168,36 @@ struct TrainingConfigurationResult {
 // Thin wrapper over internal C++ Optimizer
 struct PyOptimizer {
   PyOptimizer(const std::string optimizer_model_uri, onnxruntime::training::api::CheckpointState* state,
-              std::vector<std::shared_ptr<IExecutionProvider>> providers)
+              std::vector<std::shared_ptr<IExecutionProvider>> providers, PySessionOptions* session_options)
       : optimizer_() {
     auto env = GetTrainingEnv().GetORTEnv();
     // XXX: We hope that env will be around when optimizer needs it.
     optimizer_ = std::make_shared<onnxruntime::training::api::Optimizer>(
-        optimizer_model_uri, state, onnxruntime::SessionOptions(), *env, providers);
+        optimizer_model_uri, state, session_options->value, *env, providers, session_options->custom_op_domains_);
   }
 
   std::shared_ptr<onnxruntime::training::api::Optimizer> optimizer_;
 };
 #endif
 
 struct PyGradientGraphBuilder {
-  std::unique_ptr<GradientGraphBuilder> builder;
-  std::shared_ptr<Model> model;
-  std::unique_ptr<logging::Logger> logger;
-  std::unique_ptr<GradientGraphConfiguration> gradient_graph_config;
-  PyGradientGraphBuilder(std::unique_ptr<GradientGraphBuilder> builder_, std::shared_ptr<Model> model_, std::unique_ptr<logging::Logger> logger_, std::unique_ptr<GradientGraphConfiguration> gradient_graph_config_)
-      : builder(std::move(builder_)), model(std::move(model_)), logger(std::move(logger_)), gradient_graph_config(std::move(gradient_graph_config_)) {}
+  std::unique_ptr<GradientGraphBuilder> builder_;
+  std::shared_ptr<Model> model_;
+  std::unique_ptr<logging::Logger> logger_;
+  std::unique_ptr<GradientGraphConfiguration> gradient_graph_config_;
+  std::shared_ptr<CustomRegistry> custom_registry_;
+  IOnnxRuntimeOpSchemaRegistryList local_registries_;
+  PyGradientGraphBuilder(std::unique_ptr<GradientGraphBuilder> builder,
```

Comment:
`PyGradientGraphBuilder ` did not looks to be gradient graph builder only things, it is more like a `PyGradientGraphBuildContext` which has the builder, and also other configs .